dev to main BLOCKED by issue #252

smohiudd opened the pull request:
Note: #252 needs to be completed before the next merge into the main branch

ividito committed:
feat: Placeholder DAG for stactools ingest

ividito committed:
feat: Working stactools-noaa-hrrr items DAG

botanical commented on infrastructure/variables.tf:
```diff
@@ -76,6 +76,11 @@ variable "vector_vpc" {
+  type    = bool
+  default = "false"
+}
+
```
I know this is a PR to promote dev to main, but when it is necessary to write a corresponding change to the variables template file? Do we not need to do that in this case?

smohiudd replied:
yes we will need TF_VAR_deploy_vector_automation=True for staging

botanical commented on infrastructure/s3_event_bridge_lambda.tf:
```diff
@@ -0,0 +1,203 @@
+variable "DEPLOY_VECTOR_AUTOMATION" {
+  type    = bool
+  default = false
+}
```
do we need this if it's defined in the variables file? also, it's uppercase 😯 I think the idiomatic style is lowercase

smohiudd replied:
good catch! We don't need this since it's already in `variables.tf`. I'll delete it

botanical commented on infrastructure/functions/s3_event_bridge_to_sfn_execute/lambda_function.py:
```diff
@@ -0,0 +1,60 @@
+import json
+
+
+#mwaa_env_name = 'veda-pipeline-dev-mwaa'
```
```suggestion
```

botanical commented on infrastructure/s3_event_bridge_lambda.tf:
```diff
@@ -0,0 +1,203 @@
+  function_name    = aws_lambda_function.lambda[0].function_name
+  principal        = "s3.amazonaws.com"
+  statement_id     = "AllowInvocationFromS3Bucket-veda-${var.stage}"
+  source_account   = "114506680961"
```
Sorry if you explained this already but why do we need to hard code this value?

botanical reviewed (approved):
Had a couple questions but this looks good

amarouane-ABDELHAK committed:
Add S3 event bridge

amarouane-ABDELHAK committed:
Add S3 event bridge

amarouane-ABDELHAK committed:
Add S3 event bridge

amarouane-ABDELHAK committed:
Add env deployment example

amarouane-ABDELHAK committed:
Merge pull request #251 from NASA-IMPACT/feat/use-sm2a-in-eventbrige-lambda

Feat/use sm2a in eventbrige lambda

SwordSaintLancelot committed:
updated readme for the folder and resolved the failure bug

anayeaye committed:
only execute deploy action on push to dev branch

anayeaye committed:
Merge pull request #254 from NASA-IMPACT/ci/remove-staging-action

ci: remove automated deployment for staging environment

anayeaye committed:
ci: do not automatically deploy mwaa to any environment

anayeaye committed:
Merge pull request #255 from NASA-IMPACT/ci/remove-all-mwaa-deploy-actions

ci: do not automatically deploy mwaa to any environment

ividito committed:
Add placeholder transfer to dataset DAG (no-op for now, just testing control flow)
- also modified sm2a makefile to run on ubuntu

ividito committed:
More makefile adjustments, adjust control flow

amarouane-ABDELHAK committed:
Adding condition on s3 event bridge

amarouane-ABDELHAK committed:
Adding condition on s3 event bridge

amarouane-ABDELHAK committed:
Fix some missing variables

amarouane-ABDELHAK committed:
Remove .env file

amarouane-ABDELHAK committed:
Remove .env file

amarouane-ABDELHAK committed:
Remove the airflow config to be created by CICD

ividito committed:
Reorder transfer <> discovery, change transfer task input

anayeaye committed:
print all transfer exceptions

amarouane-ABDELHAK committed:
Merge pull request #257 from NASA-IMPACT/feat/use-sm2a-in-eventbrige-lambda

Adding condition on s3 event bridge

anayeaye committed:
Merge branch 'dev' into fix/reveal-transfer-exception

anayeaye committed:
print client err on failed transfer

anayeaye committed:
raise transfer exception

anayeaye committed:
update secret variable type to string and add assume role arns to airflow_custom_variables"

anayeaye committed:
Merge pull request #260 from NASA-IMPACT/fix/reveal-transfer-exception

fix(transfer-dag): add needed assume role and stop discarding exceptions

anayeaye committed:
fix(dockerfile) remove pypgstac install

anayeaye committed:
fix(dag-requirements) remove pypgstac install

anayeaye committed:
Merge pull request #262 from NASA-IMPACT/fix/remove-pypgstac

fix: remove pypgstac install

anayeaye committed:
let ingest api handle datetime validation and just pass through inputs in collection generation dag

anayeaye committed:
Merge pull request #263 from NASA-IMPACT/fix/collection-datetime-format

fix: remove python version dependent datetime formatter in collection generation

smohiudd committed:
temp print statement

smohiudd committed:
Merge pull request #264 from NASA-IMPACT/fix/collection-print

Collection generation print statement

smohiudd committed:
remove comma

smohiudd committed:
Merge pull request #265 from NASA-IMPACT/fix/collection-print

Remove comma in collection generation

ividito committed:
Strip thumbnail assets from payload before discovery

ividito committed:
rename var

ividito committed:
Update dags/veda_data_pipeline/veda_dataset_pipeline.py

Co-authored-by: Alexandra Kirk <alexandra@developmentseed.org>

ividito committed:
Fix callable

amarouane-ABDELHAK committed:
Refactor Dataset Pipeline

amarouane-ABDELHAK committed:
Refactor Dataset Pipeline

amarouane-ABDELHAK committed:
Return status of the ingest

amarouane-ABDELHAK committed:
Update dags/veda_data_pipeline/groups/discover_group.py

Co-authored-by: Jennifer Tran <12633533+botanical@users.noreply.github.com>

amarouane-ABDELHAK committed:
Adding aut ingest

amarouane-ABDELHAK committed:
Fix the bug of updating a pointer

amarouane-ABDELHAK committed:
Merge branch 'refactor-dataset-pipeline' of https://github.com/NASA-IMPACT/veda-data-airflow into refactor-dataset-pipeline

amarouane-ABDELHAK committed:
Fix updating the copy of config instead of the pointer

amarouane-ABDELHAK committed:
Merge pull request #268 from NASA-IMPACT/refactor-dataset-pipeline

Refactor Dataset Pipeline

amarouane-ABDELHAK committed:
Fixing automation DAG

amarouane-ABDELHAK committed:
Fixing automation DAG

amarouane-ABDELHAK committed:
Fixing automation DAG

amarouane-ABDELHAK committed:
Add slack notification libraries

amarouane-ABDELHAK committed:
Add slack notification libraries

amarouane-ABDELHAK committed:
Merge pull request #236 from NASA-IMPACT/feature/optimize_automated_cog_transformation

PI objectives for the transformation pipeline

amarouane-ABDELHAK committed:
Update SM2A workers python version to 3.11

amarouane-ABDELHAK committed:
Add SM2A RBAC to SM2A

amarouane-ABDELHAK committed:
Update the CMD for the workers

amarouane-ABDELHAK committed:
Add newline

amarouane-ABDELHAK committed:
Fix some formatting

amarouane-ABDELHAK committed:
Merge pull request #271 from NASA-IMPACT/fix/update-workers-python-version

Update SM2A workers python version to 3.11

ividito committed:
Break out promotion to new DAG

ividito committed:
Remove transfer flag

ividito committed:
Light refactors to shared tasks, fixes to align promotion to dataset refactors

ividito committed:
set transfer dry run default back to false

ividito committed:
Fix makefile, fix ti KeyError

siddharth0248 committed:
regex updated

siddharth0248 committed:
Merge pull request #274 from NASA-IMPACT/regex/sid

Regex updated

ividito committed:
fix(transfer): add non-decorated shared transfer function

ividito committed:
Remove defunct comments

ividito committed:
Terraform lock update

ividito committed:
Update terraform actrion version

ividito committed:
fix(submit-stac): modify retry behavior for API submission

ividito committed:
fix(build_stac): Use airflow vars instead of env var for build stac role

ividito committed:
oops

ividito committed:
Missing Variable import

ividito committed:
Add .airflowignore

botanical committed:
fix: add pystac stac version override

botanical committed:
fix: change condition in remove_thumnail_asset to resolve key error for assets

botanical committed:
fix: update changelog

botanical committed:
fix: revert changelog changes

botanical committed:
Merge pull request #292 from NASA-IMPACT/issue-291-support-stac-v1

fix: add pystac stac version override

amarouane-ABDELHAK committed:
Automate the transformation

amarouane-ABDELHAK committed:
Fix check function

amarouane-ABDELHAK committed:
fix base URL

amarouane-ABDELHAK committed:
revert base URL

amarouane-ABDELHAK committed:
Updating a variable name

paridhi-parajuli committed:
add raw data filter regex and task

paridhi-parajuli committed:
add config param as str

paridhi-parajuli committed:
increase runs per dag

paridhi-parajuli committed:
1. Combine filter task to discover files task
2. Make xcom to pass the s3 location of discovered files rather than the list itself
3. Do chunking for writing the files to s3 if its more than 900

botanical committed:
fix: update to fail safely instead

botanical committed:
Merge pull request #293 from NASA-IMPACT/issue-287/resolve-key-error-assets

fix: Updates to fail gracefully if there are no assets to resolve key_error

paridhi-parajuli committed:
resolve comment

paridhi-parajuli committed:
add new line

paridhi-parajuli committed:
make the max instances per dag configurable

paridhi-parajuli committed:
add changes to temp file handling

amarouane-ABDELHAK committed:
Merge pull request #294 from NASA-IMPACT/automated-transformation/add-raw-data-filter

Add raw data filter regex and task

amarouane-ABDELHAK committed:
Add xcom devision logic

amarouane-ABDELHAK committed:
Add check function

paridhi-parajuli committed:
1. Remove generic vector ingest dag
2. Add more dag configs
3. Modigy the ogr2ogr command
4. Take id template and filename for layername when collection is not provided

paridhi-parajuli committed:
remove generic vector ingest

paridhi-parajuli committed:
add default values

paridhi-parajuli committed:
resolve bug

botanical committed:
fix: use configuration dict passed into dag run for transfer_data

paridhi-parajuli committed:
allow dag to fail if even one file processing fails

paridhi-parajuli committed:
pass status

botanical committed:
fix: update cogify type to be a boolean

paridhi-parajuli committed:
resolve loop termination bug

paridhi-parajuli committed:
handle failure report

paridhi-parajuli committed:
add rds disaster recovery

paridhi-parajuli committed:
add docs

botanical committed:
fix: add explicit type for dry_run, set default for cogify

botanical committed:
fix: add default value for dry_run

botanical committed:
Merge pull request #301 from NASA-IMPACT/jt/issue-295-fix-valid-bucket-error

fix: use configuration dict passed into dag run for transfer_data

paridhi-parajuli committed:
update readme with new dag configs

paridhi-parajuli committed:
add roles

paridhi-parajuli committed:
add variables for rds disaster recovery

paridhi-parajuli committed:
add delete glue db task

paridhi-parajuli committed:
new line

ividito committed:
feat: Split off noaa example, add generic stactools pipeline

paridhi-parajuli committed:
Merge pull request #298 from NASA-IMPACT/update/vector-ingestion

Update vector ingestion

paridhi-parajuli committed:
add rds disaster recovery in sm2a

paridhi-parajuli committed:
get roles that are already created through aws_iam_role

paridhi-parajuli committed:
remove duplicate vars

paridhi-parajuli committed:
1.
a. correct the key for secrets retrieval
b. create a s3 bucket

paridhi-parajuli committed:
resolve bug

botanical committed:
feat: update docs and example env files

botanical committed:
fix: clean up env files

paridhi-parajuli committed:
temporarily remove delete glue db task

paridhi-parajuli committed:
undo:temporarily remove delete glue db task

paridhi-parajuli committed:
Merge pull request #304 from NASA-IMPACT/feature/automate-to-cog-transformation

Update automated cog transformation dag

paridhi-parajuli committed:
remove delete task

paridhi-parajuli committed:
undo prev commit

ividito committed:
fix: Rework airflow container requirements loading

amarouane-ABDELHAK committed:
Add invalidation to vector ingest DAG

botanical committed:
fix: revert changes to sm2a-local-config/env_example

botanical committed:
Merge pull request #306 from NASA-IMPACT/issue-303/update-docs

feat: update docs and example env files

ividito committed:
fix: item ingest without payload file, dry run disabled

amarouane-ABDELHAK committed:
Fix policy

amarouane-ABDELHAK committed:
Fix policy

amarouane-ABDELHAK committed:
Update readme

amarouane-ABDELHAK committed:
Update readme

amarouane-ABDELHAK committed:
Merge pull request #310 from NASA-IMPACT/fix/update-readme

Update readme

anayeaye committed:
feat: generate different types of scheduled dags

anayeaye committed:
comment transfer dag generation until configured

anayeaye committed:
use existing vector naming pattern

anayeaye committed:
log vector ingest object

anayeaye committed:
remove comments

anayeaye committed:
Merge pull request #311 from NASA-IMPACT/feat/schedule-vector-ingest

feat: schedule multitple dag types

anayeaye committed:
fix: explicitly add schedule interval instead of trusting template

anayeaye committed:
Merge pull request #312 from NASA-IMPACT/fix/generate-vector-dag-schedule

fix: explicitly add schedule interval instead of trusting template

anayeaye committed:
fix: only set vector schedule once on dag generation

anayeaye committed:
committed wrong sm2a/dag instead of dag

anayeaye committed:
Merge pull request #313 from NASA-IMPACT/fix/vector-dag-schedule-arg

fix: only set vector schedule once on dag generation

anayeaye committed:
require named parameters for clarity

anayeaye committed:
Merge branch 'dev' into fix/dynamic-dag-args

anayeaye committed:
revert unintented changes

anayeaye committed:
revert unintented changes

anayeaye committed:
Update dags/veda_data_pipeline/utils/s3_discovery.py

anayeaye committed:
Update dags/veda_data_pipeline/groups/discover_group.py

anayeaye committed:
require prefix in discovery event run config

anayeaye committed:
Merge pull request #314 from NASA-IMPACT/fix/dynamic-dag-args

Fix/dynamic dag args

anayeaye committed:
fix: adjust signature to accept hidden airflow kwargs

anayeaye committed:
Merge pull request #316 from NASA-IMPACT/fix/dynamic-dag-generator-args

fix: adjust signature to accept hidden airflow kwargs

anayeaye committed:
fix: adjust s3 discovery group signature to accept hidden airflow kwargs

anayeaye committed:
Merge pull request #317 from NASA-IMPACT/fix/dag-import-errors-kwargs

fix: adjust s3 discovery group signature to accept hidden airflow kwargs

anayeaye committed:
fix: vector ingest can use href from object metadata that has already been preprocessed for cog ingest

anayeaye committed:
extract filename in one step

anayeaye committed:
Merge pull request #319 from NASA-IMPACT/fix/vector-ingest-use-custom-keys

fix: vector ingest can use href from preprocessed object metadata

anayeaye committed:
replace deprecated get_files_to_process with get_files_task

anayeaye committed:
Merge pull request #320 from NASA-IMPACT/fix/vector-get-files-task

replace deprecated get_files_to_process with get_files_task

amarouane-ABDELHAK committed:
Only invalidate

amarouane-ABDELHAK committed:
Add CF invalidation

botanical committed:
fix: update workflows to use full commit hash

paridhi-parajuli committed:
remove snapshot bucket creation

paridhi-parajuli committed:
revert back

paridhi-parajuli committed:
...

paridhi-parajuli committed:
Merge branch 'dev' into add/rds-disaster-recovery

amarouane-ABDELHAK committed:
Fixed conflict

paridhi-parajuli committed:
add vector ingestion readme

paridhi-parajuli committed:
Merge branch 'dev' into add/rds-disaster-recovery

paridhi-parajuli committed:
Merge pull request #324 from NASA-IMPACT/add/vector-ingest-readme

Add vector ingestion readme

anayeaye committed:
discover_from_s3 requires event or config object

anayeaye committed:
Update dags/veda_data_pipeline/veda_vector_pipeline.py

anayeaye committed:
Merge pull request #326 from NASA-IMPACT/fix/regression-in-vector-ingest

fix(ingest-vector-dag): discover_from_s3 requires event or config object

anayeaye committed:
use event config to set invalidate_cloudfront

anayeaye committed:
Update dags/veda_data_pipeline/veda_vector_pipeline.py

anayeaye committed:
Merge pull request #328 from NASA-IMPACT/fix/invalidate-cloudfront-params

use event config to set invalidate_cloudfront

amarouane-ABDELHAK committed:
Running Discover and vector DAG without Template

anayeaye committed:
Update dags/veda_data_pipeline/veda_vector_pipeline.py

amarouane-ABDELHAK committed:
Merge pull request #329 from NASA-IMPACT/fix/run-discover-and-vector-dag-without-template

Running Discover and vector DAG without Template

anayeaye committed:
convert params to dict

anayeaye committed:
fix default

anayeaye committed:
keep params until actually triggered

anayeaye committed:
Merge pull request #330 from NASA-IMPACT/fix/run-conf-airflow-params-dict

Fix/run conf airflow params dict

paridhi-parajuli committed:
added note about snapshot bucket creation

paridhi-parajuli committed:
Merge pull request #302 from NASA-IMPACT/add/rds-disaster-recovery

Add RDS disaster recovery

ividito committed:
Revert "Add RDS disaster recovery"

ividito committed:
Revert "Revert "Add RDS disaster recovery""

This reverts commit 448033cfcf3a877d83ac5bb2770ca118e5a540f0.

anayeaye committed:
generated dag params should be template conf which are overwritten when an event json is submitted

paridhi-parajuli committed:
1. Make the snapshot bucket default to empty
2. Fix the invalid policy

ividito committed:
Fix TF syntax

anayeaye committed:
Merge pull request #335 from NASA-IMPACT/fix/generated-dag-params

generated dag params should be template conf which are overwritten wh…

anayeaye committed:
allow empty optional datetime range in discovery config

anayeaye committed:
allow None in optional datetime range enums

anayeaye committed:
Merge pull request #336 from NASA-IMPACT/fix/bad-param-typing

fix(dag-generation): allow empty optional datetime_range

anayeaye committed:
allow empty string in optional datetime range enums

ividito committed:
fix: make rds backup functionality optional

botanical committed:
Merge pull request #323 from NASA-IMPACT/issue-569/commit-hash-gh-actions

fix: update workflows to use full commit hash

ividito committed:
fix: change CSDAP tag

ividito committed:
fix: review changes, condense rds_backup outputs

botanical committed:
feat: upgrade sm2a to use v1.1.7

botanical committed:
Merge pull request #339 from NASA-IMPACT/feat/upgrade-sm2a-zip-version

feat: upgrade sm2a to use v1.1.7

botanical committed:
fix: when transfer is false in a promotion config, do not transfer files

botanical committed:
fix: remove default since it's already set

ividito committed:
Rebase + direct stac submission workaround

botanical committed:
fix: add unit tests for testing promotion dag

botanical committed:
fix: follow convention and check value in payload and then check dag run conf

botanical committed:
fix: update default transfer to be true

botanical committed:
fix: update to not throw error when object is up to date

botanical committed:
fix: update unit tests

botanical committed:
fix: add unit test for 412 error

botanical committed:
fix: update unit tests to use moto instead

botanical committed:
fix: add moto to test requirements

botanical committed:
fix: remove unnecessary ti vars and use EmptyOperator since DummyOperator is deprecated

botanical committed:
fix: move transfer condition to transfer_assets_to_production_bucket function

botanical committed:
fix: update deprecated objects

botanical committed:
fix: use absolute import path for stac.py

botanical committed:
fix: test_s3_discovery

botanical committed:
fix: update test_submit_stac unit test to mock realistic ingestions result

botanical committed:
Merge pull request #342 from NASA-IMPACT/issue-300/fix-transfer

fix: when transfer is false in a promotion config, do not transfer files. when files are up to date, continue

botanical committed:
Merge branch 'dev' into issue-344/update-deprecated-objs

ividito committed:
Re-enable github, add airflow constraint

ividito committed:
Use bitshift over upstream/downstream calls

botanical committed:
Merge pull request #346 from NASA-IMPACT/issue-344/update-deprecated-objs

fix: update deprecated objs and fix unit tests

ividito committed:
Add uv to project root, set up initial dependencies

amarouane-ABDELHAK committed:
Trigger DAG via lambda invoke

ividito committed:
Revise constraints to keep xmlsec from updating

ividito committed:
Revise constraints to keep xmlsec from updating

ividito committed:
Update tests, add more test dependencies

amarouane-ABDELHAK committed:
Merge branch 'dev' into feature/trigger-dag-using-lambda-invoke

amarouane-ABDELHAK committed:
Merge branch 'fix/pin-xmlsec' into feature/trigger-dag-using-lambda-invoke

amarouane-ABDELHAK committed:
Merge branch 'dev' into feature/trigger-dag-using-lambda-invoke

amarouane-ABDELHAK committed:
Add cloudwatch logging

amarouane-ABDELHAK committed:
Merge pull request #355 from NASA-IMPACT/feature/trigger-dag-using-lambda-invoke

Trigger DAG via lambda invoke

ividito committed:
fix: keep deploy-requirements.txt until CI is updated

amarouane-ABDELHAK committed:
Merge branch 'dev' into feature/add-sm2a-rbac-to-dags

amarouane-ABDELHAK committed:
Merge pull request #272 from NASA-IMPACT/feature/add-sm2a-rbac-to-dags

Add SM2A RBAC to SM2A
